bliss/catalog.py

```diff
@@ -451,33 +436,17 @@ def galaxy_bools(self) -> Tensor:
         assert is_galaxy.size(2) == 1
         return is_galaxy * self.is_on_mask.unsqueeze(2)
 
-    def on_fluxes(self, unit: str):
-        match unit:
-            case "nmgy":
-                return self.on_nmgy
-            case "mag":
-                return self.on_mag
-            case "njymag":
-                return self.on_njymag
-            case _:
-                raise NotImplementedError()
-
     @property
-    def on_nmgy(self) -> Tensor:
+    def on_fluxes(self) -> Tensor:
         # ideally we'd always store fluxes rather than star_fluxes and galaxy_fluxes
         if "galaxy_fluxes" not in self:
             fluxes = self["star_fluxes"]
         else:
             fluxes = torch.where(self.galaxy_bools, self["galaxy_fluxes"], self["star_fluxes"])
         return torch.where(self.is_on_mask[..., None], fluxes, torch.zeros_like(fluxes))
 
-    @property
-    def on_mag(self) -> Tensor:
-        return convert_nmgy_to_mag(self.on_nmgy)
-
-    @property
-    def on_njymag(self) -> Tensor:
-        return convert_nmgy_to_njymag(self.on_nmgy)
+    def on_magnitudes(self, c) -> Tensor:
+        return convert_flux_to_magnitude(self.on_fluxes, c)
```

Comment:
Where is `c` defined? I believe `c` is survey specific.

Reply:
I tried to modify the `Survey` class but find that we also need to change the `Encoder` and some notebooks. So I move the zero point specification to our metrics by adding one argument `mag_zero_point` to our metric classes.